### PR TITLE
[Fix] Reordering tabular platform tests to resolve hanging

### PR DESCRIPTION
**Pull request by prateekdesai04**
*Description of changes:*
Platform tests were hanging because there is an issue in **_tabular/tests/unittests/resource_allocation/test_resource_allocation_combined.py_** 
Re-ordering the tests fixes that
Link to platform test run: https://github.com/prateekdesai04/autogluon/actions/runs/8533360155
Link to platform test [updated] after refactoring: https://github.com/prateekdesai04/autogluon/actions/runs/8534769943

Note that it fails for Python 3.11 and doesn't seem to be a platform (OS) issue per-se, so please proceed with merging once reviewed.
We can hope that these might go once Ray is upgraded.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |
| types-requests==2.31.0.20240402 | types-requests==2.31.0.20240403 |
| types-requests==2.31.0.20240402 | types-requests==2.31.0.20240403 |

File: .github/workflow_scripts/test_tabular.sh [hunk 1]
```diff
@@ -28,7 +28,9 @@ fi
 cd tabular/
 if [ -n "$ADDITIONAL_TEST_ARGS" ]
 then
-    python -m pytest --junitxml=results.xml --runslow "$ADDITIONAL_TEST_ARGS" tests
+    # TODO: Temporary re-arrangement of tests to unblock timeout on platform tests
```
Comment (tonyhoo) [inline .github/workflow_scripts/test_tabular.sh:31]: A more effective method would be to utilize hooks from conftest.py, available [here](https://github.com/autogluon/autogluon/blob/ab9f9e1762453d7a4984227addfc91758f772048/tabular/tests/conftest.py#L50), to conduct specific tests by order. While I wouldn't advise this as a permanent solution, it is sufficient to proceed with our 1.1 release. In order to follow up on this, please create an issue for tracking comprehensive fixes in the future.

**Review by AnirudhDagar — APPROVED** [review]
Thanks @prateekdesai04, yes the 3.11 failures will go away with ray upgrade.

**Comment by prateekdesai04** [comment]
Thanks @tonyhoo for help with refactoring!

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

**Comment by AnirudhDagar** [comment]
Platform tests are still timing out https://github.com/prateekdesai04/autogluon/actions/runs/8534769943/job/23379813975

**Comment by AnirudhDagar** [comment]
Did the platform tests pass, not sure if we were supposed to merge this already?

**Pull request merged**